src/map_inner.rs

```diff
@@ -300,30 +316,47 @@ where
                                 new_slot.into_box();
                             }
 
-                            return WriteResult::Succ(Self::unwrap_slot(target_slot).1);
+                            return InsertResult::Fail(Self::unwrap_slot(target_slot).1);
                         }
                         if slot_idx.tbl_idx != 0 {
                             // GetOrInsert only inserts key-value pair into the primary slot.
                             // So if the primary slot is not empty, we force trigger a relocation.
                             need_relocate = true;
                         }
                     }
-                    InsertType::CompareAndUpdate(old_value, compare_func) => {
+                    InsertType::CompareAndUpdate(old_value, compare_fn) => {
                         // The insert type is `compareAndUpdate`, so we need to check if
                         // the current value equals to the old_v.
+                        if !key_exist {
+                            unsafe {
+                                new_slot.into_box();
+                            }
```

Comment:
I prefer the "explict drop" style to "implict drop". 
clippy restriction lints: `clippy::undocumented_unsafe_blocks`
```rust
// SAFETY: {{explain why we can drop the `new_slot`}}
unsafe{ drop(new_slot.into_box()) };
```

Reply:
The "implict drop" can be checked by adding `#[must_use]` to the `into_box` method.

Reply:
Done